handler/enqueue_annotation.go

```diff
@@ -80,8 +117,9 @@ func (e *EnqueueRequestForAnnotation) Generic(evt event.GenericEvent, q workqueu
 	}
 }
 
+// getAnnotationRequests checks if the provided object has the annotations so as to enqueue the reconcile request.
 func (e *EnqueueRequestForAnnotation) getAnnotationRequests(object metav1.Object) (bool, reconcile.Request) {
-	if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type {
+	if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type.String() {
```

Comment:
Nit: annotations may be nil, although I doubt this will happen often.
```suggestion
	if len(object.GetAnnotations()) == 0 {
		return false, reconcile.Request{}
	}

	if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type.String() {
```